GUIDE.md

```diff
@@ -177,15 +177,22 @@ After recursive search, ripgrep's most important feature is what it *doesn't*
 search. By default, when you search a directory, ripgrep will ignore all of
 the following:
 
-1. Files and directories that match the rules in your `.gitignore` glob
-   pattern.
+1. Glob patterns specified in several ignore files:
+   1. `.gitignore`, only in git repos.
+   2. `.ignore`, which for conflicting rules takes precedence over `.gitignore`.
+   3. `.rgignore` with the highest precedence.
```

Comment:
I think saying that `.rgignore` has the highest precedence isn't quite correct. And having this in list form kind of implies that it is exhaustive. But it's not. For example, this leaves out global gitignores and `.git/info/exclude` rules. It also leaves out ignore files explicitly specified on the command line via `--ignore-file`.

This is kind of why I wrote the guide the way I did, because being 100% precise here ends up making it quite a bit more verbose and more like a reference than a guide. The guide is definitely not meant to be reference. It is explicitly imprecise at the expense of leaving out every detail but at the benefit of giving users more succinct material to read.

I do think your PR here straddles a decent line. So instead of trying to make this complete, I'd suggest some phrasing changes. So instead of "Glob patterns specified in several ignore files:," I'd say, "Files and directories that match glob patterns in these three categories:" Then I'd list it like this:

1. gitignore globs (including global and repo-specific globs)
2. `.ignore` globs, which take precedence over all gitignore globs when there's a conflict.
3. `.rgignore` globs, which take precedence over all `.ignore` globs when there's a conflict.